Session_2/module_2.py

```diff
@@ -1,35 +1,89 @@
-# from collections import defaultdict as dd
-# from itertools import product
+
 from typing import Any, Dict, List, Tuple
 
 
 def task_1(data_1: Dict[str, int], data_2: Dict[str, int]):
-    pass
+    combined_dict = data_1.copy()
+
+    for key, value in data_2.items():
+        if key in combined_dict:
+            combined_dict[key] += value
+        else:
+            combined_dict[key] = value
+
+    return combined_dict
 
 
 def task_2():
-    pass
+    result = {i: i ** 2 for i in range(1, 16)}
+    return result
 
 
 def task_3(data: Dict[Any, List[str]]):
-    pass
+    def generate_combinations(key, current_combination):
```

Comment:
Good solution!

Do you know which python function can generate combinations of sequences?

Reply:
combinations function